### ISSUE TITLE
E1019 Throwing False Error When GetAtt DBClusterResourceId

### ISSUE THREAD

**Issue by jayzandme** [aws-cloudformation/cfn-lint#2400]
### CloudFormation Lint Version

cfn-lint 0.66.1

### What operating system are you using?

Mac

### Describe the bug

When running cfn-lint, we get the following error
```
E1019 Parameter TestMySQLDbCluster.DBClusterResourceId for Fn::Sub not found at Resources/TestMySQLReadOnlyPolicy/Properties/PolicyDocument/Statement/0/Resource/Fn::Sub
templatepath/test.yaml:30:13
```
However, based on the aws docs and after ignoring the cfn-lint error, this seems like it should be valid.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#aws-resource-rds-dbcluster-return-values

### Expected behavior

Expecting the error not to be thrown

### Reproduction template

```yaml
Parameters:
  Username:
    Type: String

  Password:
    Type: String
    NoEcho: true

Resources:
  TestMySQLDbCluster:
    Type: AWS::RDS::DBCluster
    DeletionPolicy: Retain
    UpdateReplacePolicy: Retain
    Properties:
      Engine: "5.7.mysql_aurora.2.07.7"
      MasterUsername: !Ref Username
      MasterUserPassword: !Ref Password

  TestMySQLReadOnlyPolicy:
    Type: AWS::IAM::ManagedPolicy
    DependsOn: TestMySQLDbCluster
    Properties:
      Path: /
      PolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Action:
              - rds-db:connect
            Resource: !Sub "arn:aws:rds-db:${AWS::Region}:${AWS::AccountId}:dbuser:${TestMySQLDbCluster.DBClusterResourceId}/iam-readonly"
```

**Comment by PatMyron** [comment]
similar recent RDS attribute issue: https://github.com/aws-cloudformation/cfn-lint/issues/2343

**Comment by kddejong** [comment]
From the registry.
```
readOnlyProperties": [
    "/properties/DBClusterResourceId",
    "/properties/Endpoint",
    "/properties/Endpoint/Address",
    "/properties/Endpoint/Port",
    "/properties/ReadEndpoint/Port",
    "/properties/ReadEndpoint/Address"
  ],
```

**Comment by kddejong** [comment]
Going to have to patch this in.  Was hoping it made it in to the last spec but its still not there.
